cypress/utils/apps/data_explorer/index.d.ts

```diff
@@ -19,5 +19,33 @@ declare namespace Cypress {
     clearSaveQuery(): Chainable<any>;
     deleteSaveQuery(name: string): Chainable<any>;
     switchDiscoverTable(name: string): Chainable<any>;
+    getDocTableRow(rowNumber: number): Chainable<any>;
+    getDocTableField(columnNumber: number, rowNumber: number): Chainable<any>;
+    findExpandedDocTableRows(expandedDocument: Chainable<any>): Chainable<any>;
+    getExpandedDocTableRow(
+      docTableRowNumber: number,
+      expandedDocumentRowNumber: number
+    ): Chainable<any>;
+    getExpandedDocTableRowValue(
+      docTableRowNumber: number,
+      expandedDocumentRowNumber: number
+    ): string;
+    getExpandedDocTableRowFieldName(
+      docTableRowNumber: number,
+      expandedDocumentRowNumber: number
+    ): string;
+    selectIndexDatasetLanguage(datasetLanguage: string);
+    selectIndexDataset(indexClusterName: string, indexName: string, datasetLanguage: string);
+    selectIndexPatternDatasetLanguage(datasetLanguage: string);
+    selectIndexPatternDataset(indexPatternName: string, datasetLanguage: string);
+    toggleDocTableRow(rowNumber: number);
+    verifyDocTableFilterAction(
+      rowNumber: number,
+      filterElement: string,
+      expectedQueryHitsWithoutFilter: string,
+      expectedQueryHitsAfterFilterApplied: string,
+      shouldMatch: boolean
+    );
+    verifyDocTableFirstExpandedFieldFirstRowToggleColumnButtonHasIntendedBehavior();
```

Comment:
Because these aren't Cypress custom commands, the types shouldn't be added here nested within `declare namespace Cypress { interface Chainable`. Because then our IDEs think we can chain these off of `cy`:

![Uploading Screenshot 2024-12-31 at 1.08.53 PM.png…]()

If you move these functions to a separate file, we don't need to create a typescript dec file for it, since you already JSDoc-annotated it